### PR TITLE
Marketplace resource tests

### PR DESCRIPTION
**Pull request by tyleryasaka**
### Checklist:

- [x] Code contains relevant tests for the problem you are solving
- [x] Ensure all new and existing tests pass
- [x] Update any relevant READMEs and [docs](https://github.com/OriginProtocol/docs)

### Description:

Adds tests for the marketplace resource

### CODE REVIEW

File: test/service-data-store.test.js [hunk 1]
```diff
@@ -24,7 +29,8 @@ describe('ListingIpfsStore load', () => {
   })
 
   it(`Should load a valid object`, async () => {
-    mockIpfsService.loadObjFromFile = sinon.stub().resolves(goodListing)
+    console.log('good listing', Object.assign({}, goodListing))
+    mockIpfsService.loadObjFromFile = sinon.stub().resolves(Object.assign({}, goodListing))
```
Comment (franckc) [inline test/service-data-store.test.js]: That is so weird ! Why do we need to clone this object ???

File: src/index.js [hunk 2]
```diff
@@ -17,6 +17,7 @@ const defaultIpfsApiPort = '5002'
 const defaultIpfsGatewayPort = '443'
 const defaultIpfsGatewayProtocol = 'https'
 const defaultAttestationServerUrl = `${defaultBridgeServer}/api/attestations`
+// const defaultIndexingServerUrl = `${defaultBridgeServer}/api`
```
[the remark this answers is not in the document]
Comment (tyleryasaka) [reply src/index.js]: Yeah I was thinking we'll add them back soon for indexing so I just commented them out. https://github.com/OriginProtocol/origin-js/pull/436/commits/fb4e7c9b5fd9f4c090308f41cad94ad21eee19c1

But could also just remove those lines.

File: test/resource_marketplace.test.js [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+import contractServiceHelper from './helpers/contract-service-helper'
+import IpfsService from '../src/services/ipfs-service.js'
+import { expect } from 'chai'
+import Web3 from 'web3'
+import listingValid from './fixtures/listing-valid.json'
+import offerValid from './fixtures/offer-valid.json'
+import reviewValid from './fixtures/review-valid.json'
+
+// oddly changing an imported object here can affect other or subsequent tests that import the same file
+const listingData = Object.assign({}, listingValid)
```
[the remark this answers is not in the document]
Comment (tyleryasaka) [reply test/resource_marketplace.test.js]: This is how I fixed the weird behavior where using imported json was causing side effects between tests.

File: test/service-data-store.test.js [hunk 4]
```diff
@@ -24,7 +29,8 @@ describe('ListingIpfsStore load', () => {
   })
 
   it(`Should load a valid object`, async () => {
-    mockIpfsService.loadObjFromFile = sinon.stub().resolves(goodListing)
+    console.log('good listing', Object.assign({}, goodListing))
```
[the remark this answers is not in the document]
Comment (tyleryasaka) [reply test/service-data-store.test.js]: oops! thanks

File: test/service-data-store.test.js (hunk 1 above)
Comment (tyleryasaka) [reply test/service-data-store.test.js]: I am not sure why yet - very curious though

**Pull request merged**